### PR TITLE
Feature immersive content

### PR DESCRIPTION
**Pull request by faloi**
This adds some handy methods that will be used for mumuki/mumuki-laboratory#1489.

### CODE REVIEW

**Review by luchotc — APPROVED** [review]
Apart from that minor comments this one looks ok.
It feels a bit weird that we are making content and content containers interface bigger by making all of them respond to those methods.
I'll look into labo counterpart of this PR and I'll check if we can't avoid this change here by changing something there!

File: app/models/user.rb [hunk 1]
```diff
@@ -243,14 +243,21 @@ def current_organic_context
...
+  def current_immersive_context_at(path_item)
     if Organization.current?
-      immersive_organization_at(exercise) || Organization.current
+      immersive_organization_at(path_item) || Organization.current
     else
       main_organization
     end
   end
 
+  def current_immersive_context_and_content_at(path_item)
```
[the remark this answers is not in the document]
Comment (faloi) [reply app/models/user.rb:254]: I like it. :ok_hand:

File: lib/mumuki/domain/helpers/user.rb [hunk 2]
```diff
@@ -100,12 +100,21 @@ def immersive_organization_at(path_item, current = Organization.current)
...
     usage_filter = path_item ? lambda { |it| path_item.used_in?(it) } : lambda { |_| true }
-    student_granted_organizations
-      .select { |it| current.immersed_in?(it) }
-      .select(&usage_filter)
+    immersive_organizations_for(current).select(&usage_filter)
+  end
+
+  def immersive_organization_with_content_at(path_item, current = Organization.current)
+    orga = immersive_organizations_with_content_at(path_item, current).single
+    [orga, orga ? path_item&.navigable_content_in(orga) : nil]
```
[the remark this answers is not in the document]
Comment (faloi) [reply lib/mumuki/domain/helpers/user.rb:109]: Well, I couldn't find a case where this `if` matters, so I'm removing it.

File: lib/mumuki/domain/helpers/user.rb [hunk 3]
```diff
@@ -119,4 +128,12 @@ def profile_fields
       [:first_name, :last_name, :gender, :birthdate]
     end
   end
+
+  private
+
+  def immersive_organizations_for(current)
```
[the remark this answers is not in the document]
Comment (faloi) [reply lib/mumuki/domain/helpers/user.rb:134]: Agree.

**Pull request merged**